lib/Rex/Commands.pm

```diff
@@ -1059,7 +1059,7 @@ sub needs {
   if ($self) {
     my $suffix = $self;
     $suffix =~ s/::/:/g;
-    @args = map { $_ = "$suffix:$_" } @args;
+    @args = map { $_ = "$suffix:$_" } @args; ## no critic ProhibitMutatingListFunctions
```

Comment:
The code as it is there now, is equivalent to:

```perl
@args = map { "$suffix:$_" } @args;
```

due to the fact that the return value of `map` is assigned to the variable being iterated over. I think it's better, even for the scope of this PR not to suppress but to fix this instance.